### DISCUSSION TITLE
How to Use React Suite with tailwind css in Next.js?

### DISCUSSION THREAD

**Discussion by KareemMamdouh** [rsuite/rsuite#1971] [Q&A]
I want to use React Suite and tailwind so I install tailwind following the installation instructions from the documentation and it's working fine, but when I install React Suite and add some configuration in the next.config.js file tailwind is no longer working, and React Suite workers well, I think the problem is with this file I hope you can help me with this problem

 **Code** 
[the example in code sandbox](https://codesandbox.io/s/peaceful-napier-y10n2)

 **stackoverflow** 
[the example in stackoverflow](https://stackoverflow.com/questions/69457456/how-to-use-react-suite-with-tailwind-css-in-next-js)

**Comment by simonguo** [comment]
https://github.com/takeItIzzy/tailwind_cli_demo
You can refer to this sample project. It should help you.

**Reply by KareemMamdouh** [reply]
When I add the CSS it works in dev mode, but when I build and run, the start as a server doesn't work any style from the CSS file

**Reply by takeItIzzy** [reply]
Hi there, here's the build product of this demo: https://tailwind-cli-demo-lkctf3d65-takeitizzy.vercel.app/ . It works well online.
This demo uses tailwind cli instead of the usual tailwind installation, please check the following steps to make sure the cli works well with nextjs:
1. Run `npx tailwindcss -o public/tailwind.css --watch`, this command line will generate a css file in public folder at first time, and update this file each time you edit `className` in react components.
2. link this css file at `_document.js`. You can copy my code [here](https://github.com/takeItIzzy/tailwind_cli_demo/blob/main/pages/_document.js).

In fact, the problem you are experiencing has nothing to do with rsuite, it is because you overrided the css webpack config built in nextjs, see [here](https://nextjs.org/docs/messages/built-in-css-disabled).

Finally, I recommend that you don't have to use preprocessors since you're using tailwindcss.

**Comment by SevenOutman** [comment]
Hi Kareem, you can import RSuite styles from compiled CSS files instead of Less source files if you don't need to write Less in your app:

```diff
- import 'rsuite/styles/index.less';
+ import 'rsuite/dist/rsuite.css';
```

That way you don't have to tweak your webpack config for Less support, which seems to be the reason why Next.js's built-in CSS support is disabled.

**Reply by KareemMamdouh** [reply]
I already did this solution temporarily before, but I wanted to use (less) to Custom Theme Colors as documentation mention here .. https://rsuitejs.com/guide/customization/
